test/parallel/test-stringbytes-external.js

```diff
@@ -119,3 +119,63 @@ var PRE_3OF4_APEX = Math.ceil((EXTERN_APEX / 4) * 3) - RADIOS;
   assert.equal(a, b);
   assert.equal(b, c);
 })();
+
+// v8 fails silently if string length > v8::String::kMaxLength
+(function() {
+  // v8::String::kMaxLength defined in v8.h
+  const kStringMaxLength = process.binding('buffer').kStringMaxLength;
+
+  assert.throws(function() {
+    var buf = new Buffer(kStringMaxLength + 1).fill().toString();
+  }, Error);
+
+  assert.throws(function() {
+    var buf = new Buffer(kStringMaxLength + 1).fill().toString('ascii');
+  }, Error);
+
+  assert.throws(function() {
+    var buf = new Buffer(kStringMaxLength + 1).fill().toString('utf8');
+  }, Error);
+
+  assert.throws(function() {
+    var buf = new Buffer(kStringMaxLength * 2 + 2).fill().toString('utf16le');
+  }, Error);
+
+  assert.throws(function() {
+    var buf = new Buffer(kStringMaxLength + 1).fill().toString('binary');
+  }, Error);
+
+  assert.throws(function() {
+    var buf = new Buffer(kStringMaxLength + 1).fill().toString('base64');
+  }, Error);
+
+  assert.throws(function() {
+    var buf = new Buffer(kStringMaxLength + 1).fill().toString('hex');
+  }, Error);
+
+  try {
+    var maxBuffer = new Buffer(kStringMaxLength * 4);
```

Comment:
Can I suggest dropping the `var maxBuffer =` here?  I don't think V8's lexical analysis pass is smart enough to figure out that this is a dead store, it'll keep a reference to the buffer alive until the function returns.  Without the assignment, it's immediately collectible.